tools.py

```diff
@@ -5,6 +5,24 @@
 
 
 class Vertex:
+    """
+    The Vertex object represents a vertex that is part of a pose graph.
+
+    Args:
+        run_id (int): Id of the run the vertex belongs to.
+        pose_id (int): Id of the pose for the vertex.
+        next_run_id (int): Id of the next consecutive run.
+        next_pose_id (int): Id of the next consecutive pose.
+        next_transform (Transform): Transform from this to the next vertex.
+        teach (bool): True if vertex belongs to a teach run, False if repeat run. 
+        prev_run_id (int): Id of the previous consecutive run (optional).
```

Comment:
Same as above. This ends up as 0 for teach vertices and -1 for repeats. Again, kind of redundant